servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -220,32 +215,36 @@ public int hashCode() {
 
     private static final class ClientFactory implements Function<UrlKey, FilterableStreamingHttpClient> {
         private static final ClientSslConfig DEFAULT_CLIENT_SSL_CONFIG = new ClientSslConfigBuilder().build();
-        private final DefaultSingleAddressHttpClientBuilder<HostAndPort, InetSocketAddress> builderTemplate;
+        private final Function<HostAndPort, SingleAddressHttpClientBuilder<HostAndPort, InetSocketAddress>>
+                builderFactory;
+        private final HttpExecutionContext executionContext;
         @Nullable
         private final SingleAddressInitializer<HostAndPort, InetSocketAddress> singleAddressInitializer;
 
-        ClientFactory(
-                final DefaultSingleAddressHttpClientBuilder<HostAndPort, InetSocketAddress> builderTemplate,
+        ClientFactory(final Function<HostAndPort, SingleAddressHttpClientBuilder<HostAndPort, InetSocketAddress>>
+                        builderFactory,
+                final HttpExecutionContext executionContext,
                 @Nullable final SingleAddressInitializer<HostAndPort, InetSocketAddress> singleAddressInitializer) {
-            this.builderTemplate = builderTemplate;
+            this.builderFactory = builderFactory;
+            this.executionContext = executionContext;
             this.singleAddressInitializer = singleAddressInitializer;
         }
 
         @Override
         public StreamingHttpClient apply(final UrlKey urlKey) {
-            // Copy existing builder to prevent changes at runtime when concurrently creating clients for new addresses
-            final HttpClientBuildContext<HostAndPort, InetSocketAddress> buildContext =
-                    builderTemplate.copyBuildCtx(urlKey.hostAndPort);
+            final SingleAddressHttpClientBuilder<HostAndPort, InetSocketAddress> builder =
+                    builderFactory.apply(urlKey.hostAndPort);
```

Comment:
Perhaps null check this result. I have noticed we have generally checked for provided Functions and Suppliers and think it is a good practice.